src/support/ios.c

```diff
@@ -831,6 +831,66 @@ size_t ios_copyuntil(ios_t *to, ios_t *from, char delim)
     return total;
 }
 
+//Copy until '\r', '\n' or '\r\n'
+size_t ios_copyline(ios_t *to, ios_t *from, int chomp)
+{
+    size_t nchomp = 0;
+    size_t total = 0, avail = (size_t)(from->size - from->bpos);
+    size_t ntowrite;
+    while (!ios_eof(from)) {
+        if (avail == 0) {
+            avail = ios_readprep(from, LINE_CHUNK_SIZE);
+            if (avail == 0)
+                break;
+        }
+        size_t written;
+
+        char *r = NULL;
+        char *n = NULL;
+
+        for (size_t i = 0; i < avail; i++){
```

Comment:
That's the first thing I tried and it's simpler and significantly slower. 

You have to look for both `\r` and `\n` in available bytes, but you only want to break on the first one found. e.g. If you have a large file and you look for `\r` both in `avail` (using memch) and never find it the code gets very slow. 

This has the same performance as `readuntil` which uses `memchr`.